packages/invoices/src/InboundShipment/DetailView/reducer/reducer.ts

```diff
@@ -0,0 +1,231 @@
+import {
+  getUnitQuantity,
+  getSumOfKeyReducer,
+  produce,
+  Column,
+  DocumentActionSet,
+  DocumentActionType,
+  SortBy,
+  ifTheSameElseDefault,
+  Item,
+  arrayToRecord,
+} from '@openmsupply-client/common';
+import { placeholderInbound } from '../../../utils';
+import {
+  ActionType,
+  OutboundShipmentAction,
+  Invoice,
+  InvoiceLine,
+  InboundShipment,
+  InboundShipmentItem,
+  InboundShipmentRow,
+} from '../../../types';
+
+const getExistingLine = (
+  items: InboundShipmentItem[],
+  line: InboundShipmentRow
+): {
+  existingSummaryItem?: InboundShipmentItem;
+  existingRow?: InboundShipmentRow;
+} => {
+  const existingSummaryItem = items.find(
+    item => !!item.batches[line.id] || item.itemId === line.itemId
+  );
+
+  if (!existingSummaryItem) return {};
+
+  let existingRow = existingSummaryItem.batches[line.id];
+
+  if (existingRow) return { existingRow, existingSummaryItem };
+
+  existingRow = Object.values(existingSummaryItem.batches).find(
+    ({ stockLineId }) => stockLineId === line.stockLineId
+  );
+
+  return { existingRow, existingSummaryItem };
+};
+
+export const recalculateSummary = (
+  summaryItem: InboundShipmentItem
+): { unitQuantity: number; numberOfPacks: number } => {
+  const unitQuantity = Object.values<InboundShipmentRow>(
+    summaryItem.batches
+  ).reduce(getUnitQuantity, 0);
+
+  const numberOfPacks = Object.values<InboundShipmentRow>(
+    summaryItem.batches
+  ).reduce(getSumOfKeyReducer('numberOfPacks'), 0);
+
+  return { unitQuantity, numberOfPacks };
+};
+
+export const InboundAction = {
```

Comment:
Using the outbound actions at the moment.. the boilerplate for reducers is too damn high!